src/coordination/coordinator_instance.cpp

```diff
@@ -198,6 +201,24 @@ auto CoordinatorInstance::GetLeaderCoordinatorData() const -> std::optional<Coor
   return raft_state_->GetLeaderCoordinatorData();
 }
 
+void CoordinatorInstance::AddOrUpdateClientConnectors(std::vector<CoordinatorToCoordinatorConfig> const &configs) {
+  auto connectors = coordinator_connectors_.Lock();
+
+  for (auto const &config : configs) {
+    if (config.coordinator_id == raft_state_->GetCoordinatorId()) {
+      continue;
+    }
+    auto const connector = std::ranges::find_if(
+        *connectors, [&config](auto &&connector) { return connector.first == config.coordinator_id; });
```

Comment:
you already have coordinator_id in CoordinatorToCoordinatorConfig saved, no need to have std::list of pairs